hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/NumOfWorkflowsToDeleteMetadataMetricActor.scala

```diff
@@ -0,0 +1,77 @@
+package cromwell.services.metadata.hybridcarbonite
+
+import java.time.OffsetDateTime
+
+import akka.actor.{ActorRef, LoggingFSM, Props, Status}
+import akka.pattern.pipe
+import cats.data.NonEmptyList
+import cromwell.core.instrumentation.InstrumentationPrefixes
+import cromwell.services.MetadataServicesStore
+import cromwell.services.instrumentation.CromwellInstrumentation
+import cromwell.services.metadata.MetadataArchiveStatus
+import cromwell.services.metadata.MetadataArchiveStatus.Archived
+import cromwell.services.metadata.hybridcarbonite.NumOfWorkflowsToDeleteMetadataMetricActor._
+import cromwell.services.metadata.impl.{MetadataDatabaseAccess, MetadataServiceActor}
+
+class NumOfWorkflowsToDeleteMetadataMetricActor(override val serviceRegistryActor: ActorRef)
+  extends LoggingFSM[MetadataDeletionMetricsActorState, Unit]
+  with MetadataDatabaseAccess
+  with MetadataServicesStore
+  with CromwellInstrumentation {
+
+  implicit val ec = context.dispatcher
+
+  startWith(WaitingForMetricCalculationRequestOrMetricValue, ())
+
+  when(WaitingForMetricCalculationRequestOrMetricValue) {
+    case Event(NumOfWorkflowsToDeleteMetadataMetricValue(value), _) =>
+      sendGauge(workflowsToDeleteMetadataMetricPath, value, InstrumentationPrefixes.ServicesPrefix)
+      stay()
+    case Event(CalculateNumOfWorkflowsToDeleteMetadataMetricValue(currentTimestampMinusDelay), _) =>
+      countRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp(
+        MetadataArchiveStatus.toDatabaseValue(Archived),
+        currentTimestampMinusDelay
+      )
+        .map(intVal => FinishedNumOfWorkflowsToDeleteMetadataMetricValueCalculation(intVal.toLong))
+        .pipeTo(self)
+      goto(MetricCalculationInProgress)
+  }
+
+  when(MetricCalculationInProgress) {
+    case Event(CalculateNumOfWorkflowsToDeleteMetadataMetricValue, _) =>
```

Comment:
You also need to handle `NumOfWorkflowsToDeleteMetadataMetricValue` in this state. 

Since it's kind of an edge case that we're still calculating a query when we get an exact number from the delete actor, I wouldn't try to be too clever about handling it (eg by cancelling our own query or changing states before our query responds). I'd probably send the new metric to the instrumentation actor but otherwise stay in the same state and wait for our existing query to give us its answer.